### ISSUE TITLE
SDK: Create separate package for exporter interface.

### ISSUE THREAD

**Issue by rghetia** [open-telemetry/opentelemetry-go#77]
as per this [comment](https://github.com/open-telemetry/opentelemetry-go/pull/65/files#r310241721).

Exporter interface is currently part of SDK. Ideally it would be nice to have a separate interface that can be reused by other SDKS.

## Requirement
- Define Export Data Types and corresponding exporter interface. E.g
```go
type ExporterType int32

const (
    ExporterTypeSync = iota
    ExporterTypeAsyncBatched,
    // For streaming SDKs
    ExporterTypeSyncEvents,
)
```
- Define Export Interface for each Data Types. E.g
```go
type SyncExporter interface {
    // implementation cannot block.
    ExportSpan(ctx context.Context, span SpanData)
}
type AsyncBatchedExporter interface {
    ExportSpans(ctx context.Context, spans []SpanData)
}
type SyncEventsExporter interface {
    // implementation cannot block.
    ExportSpanEvent(ctx context.Context, event SpanEvent) // These are streaming SDK events.
}
```
- Registry for Exporter
```go
func Register(t ExporterType, e interface{}) {
}
```

**Comment by jmacd** [comment]
Suggestion: Implement just the SyncExporter and a single Register for now?

**Comment by rghetia** [comment]
fixed by #162
